azure-pipelines.yml

```diff
@@ -36,10 +36,19 @@ jobs:
   - script: python3 -m pip install poetry && poetry install
     displayName: 'Install Project Environment'
 
+  - task: Cache@2
+    displayName: 'Restore pre-commit environment'
+    condition: and(succeeded(), eq(variables['python.version'], '3.8'))
+    inputs:
+      key: pre-commit | "$(PythonVersion.pythonLocation)" | .pre-commit-config.yaml
+      restoreKeys: |
+        pre-commit | "$(PythonVersion.pythonLocation)"
```

Comment:
The `UsePythonVersion` task needs to be given a name with `name: PythonVersion` so that this output variable can be referenced.